presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewCandidateExtractor.java

```diff
@@ -22,41 +22,49 @@
 import com.facebook.presto.sql.parser.SqlParser;
 import com.facebook.presto.sql.tree.Query;
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableMap;
 import org.testng.annotations.Test;
 
 import java.util.List;
-import java.util.Set;
+import java.util.Map;
 
 import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
 import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
 
 public class TestMaterializedViewCandidateExtractor
```

Comment:
Should we add some tests for subquery?

Reply:
Hopefully, we will soon remove this class entirely, so might not be needed.

Reply:
Initially, we wanted to use this class to fetch the most efficient materialized view in case of multiple matches. This class is super light weight at this moment.

Reply:
@jainxrohit Thanks for the context. Happy to look at implementing this in the near future in a subsequent PR.